src/webgpu/compat/api/validation/createBindGroup.spec.ts

```diff
@@ -0,0 +1,176 @@
+) {
+  if (textureBindingViewDimension) {
+    return textureBindingViewDimension;
+  }
+
+  switch (dimension) {
+    case '1d':
+      return '1d';
+    case '2d':
+    case undefined:
+      return depthOrArrayLayers > 1 ? '2d-array' : '2d';
+      break;
+    case '3d':
+      return '3d';
+  }
+}
+
+g.test('dimension_matches_textureBindingViewDimension')
+  .desc(
+    `
+    Tests that, in compat mode, the dimension of a view is compatible with a texture's textureBindingViewDimension
+    when used as a TEXTURE_BINDING.
+    `
+  )
+  .params(u =>
+    u //
+      .combine('dimension', [...kTextureDimensions, undefined])
+      .combine('textureBindingViewDimension', [...kTextureViewDimensions, undefined])
+      .combine('viewDimension', [...kTextureViewDimensions, undefined])
+      .combine('depthOrArrayLayers', [1, 2, 6])
+      .filter(
+        ({ dimension, textureBindingViewDimension, depthOrArrayLayers, viewDimension }) =>
+          textureBindingViewDimension !== 'cube-array' &&
+          isTextureBindingViewDimensionCompatibleWithDimension(
+            dimension,
+            textureBindingViewDimension
+          ) &&
+          isValidViewDimensionForDimension(dimension, depthOrArrayLayers, viewDimension) &&
+          isValidViewDimensionForDepthOrArrayLayers(
+            textureBindingViewDimension,
+            depthOrArrayLayers
+          ) &&
+          isValidDimensionForDepthOrArrayLayers(dimension, depthOrArrayLayers)
+      )
+  )
+  .fn(t => {
+    const { dimension, textureBindingViewDimension, viewDimension, depthOrArrayLayers } = t.params;
+
+    const texture = t.device.createTexture({
+      size: [1, 1, depthOrArrayLayers],
+      format: 'rgba8unorm',
+      usage: GPUTextureUsage.TEXTURE_BINDING,
+      ...(dimension && { dimension }),
+      ...(textureBindingViewDimension && { textureBindingViewDimension }),
+    } as GPUTextureDescriptor); // MAINTENANCE_TODO: remove cast once textureBindingViewDimension is added to IDL
+    t.trackForCleanup(texture);
+
+    const effectiveTextureBindingViewDimension = getEffectiveTextureBindingViewDimension(
+      dimension,
+      texture.depthOrArrayLayers,
+      textureBindingViewDimension
+    );
+
+    const effectiveViewDimension = getEffectiveTextureBindingViewDimension(
+      dimension,
+      texture.depthOrArrayLayers,
+      viewDimension
+    );
+
+    const layout = t.device.createBindGroupLayout({
+      entries: [
+        {
+          binding: 0,
+          visibility: GPUShaderStage.COMPUTE,
+          texture: {
+            viewDimension: effectiveViewDimensionForTexture(texture, viewDimension),
+          },
+        },
+      ],
+    });
+
+    const shouldError = effectiveTextureBindingViewDimension !== effectiveViewDimension;
+
+    t.expectValidationError(() => {
+      t.device.createBindGroup({
+        layout,
+        entries: [{ binding: 0, resource: texture.createView({ dimension: viewDimension }) }],
```

Comment:
It's important especially for this test to check whether the error comes from createView or createBindGroup. I'm assuming createBindGroup, so the createView should be moved out of the expectValidationError.